### PR TITLE
Add a config option to set the default build memory limit

### PR DESCRIPTION
**Pull request by Nemo157**

### CODE REVIEW

File: src/db/migrate.rs [hunk 1]
```diff
@@ -286,11 +286,16 @@ pub fn migrate(version: Option<Version>, conn: &mut Client) -> crate::error::Res
...
+            "Allow memory limits of more than 2 GB",
             // upgrade query
             "ALTER TABLE sandbox_overrides ALTER COLUMN max_memory_bytes TYPE BIGINT;",
             // downgrade query
-            "ALTER TABLE sandbox_overrides ALTER COLUMN max_memory_bytes TYPE INTEGER;"
+            "
+                ALTER TABLE sandbox_overrides
+                    ALTER COLUMN max_memory_bytes
+                    TYPE INTEGER
+                    USING least(max_memory_bytes, 2147483647);
```
Comment (syphar) [inline src/db/migrate.rs:297]: why this change? 

In the end, it would only be executed on new databases,

Comment (Nemo157) [reply src/db/migrate.rs:297]: The tests run through all the downgrades, so since I added a test that applied an override of >2GB this started failing.

Comment (Nemo157) [reply src/db/migrate.rs:297]: (It might be worth removing old migrations at some point and merging them into a higher base database schema).

**Review by syphar — APPROVED** [review]
